api/raml/core-metadata.raml

```diff
@@ -546,10 +546,12 @@ schemas:
     delete:
         description: Remove the DeviceProfile designated by database generated id. This does not remove associated commands. Returns Internal Service Error (HTTP 500) for unknown or unanticipated issues. Returns NotFoundException (HTTP 404) if the device profile cannot be found by the identifier provided. Returns DataValidationException (HTTP 413) if devices still reference the profile.
         responses:
-            "200":
-                description: boolean indicating success of the remove operation
+            "204":
```

Comment:
I'm sympathetic to changing these status codes to `204` but unfortunately that counts as an API change. If Edinburgh returns `200` for these operations and Fuji returns a `204` then Fuji isn't backward compatible. I realize both are "Success" codes but if a client is specifically looking for `200` instead of a code from the set indicating success, that client will break. Based on that, I would leave this (and the others like it) as a `200`.

Along with the `200` response we currently write out `true`. Unfortunately, that needs to be preserved as well.

RAML description updates are fine though.

Reply:
Ok, I think I made a similar change in my other PR for the update device profile. I will revert that in this PR.

Reply:
It just hurt to see `application/json` as the content type and then return "true" :frowning_face: